infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesPods.java

```diff
@@ -475,10 +476,17 @@ private ExecWatchdog() {
     }
 
     @Override
-    public void onOpen(Response response) {}
+    public void onOpen(Response response) {
+      if (response != null && response.body() != null) {
+        response.body().close();
+      }
+    }
 
     @Override
     public void onFailure(Throwable t, Response response) {
+      if (response != null && response.body() != null) {
```

Comment:
I think these workarounds are needed to avoid connection leaks. Am I right? If so, please, add comments into the code because it is not obvious at all.

Reply:
I added comments in commit https://github.com/eclipse/che/pull/8805/commits/2f9d7cf6e0fc28b099f36180b4c4303569e9ca43

Reply:
After testing again without these changes, it appears that connection leaks are still fixed (obviously by the use of the kubernetes client in API calls that use web-sockets).
So I rolled-back those changes in commit https://github.com/eclipse/che/pull/8805/commits/ddc2beb4288df376eed9c4566e923a6154d5dfbf since they are not required anymore to fix the connection leaks.